### DISCUSSION TITLE
KEDA scaling down activities

### DISCUSSION THREAD

**Discussion by cdlliuy** [kedacore/keda#2398] [Q&A / Need Help]
We run KEDA with Prometheus trigger wonderfully with http rps based metrics.
we set the scaling threshold to 100 request/s, however, in HPA side, it tries to show the actual values as 70,000m for 70 request/s.

The scaling up is good, but something is strange during the scaling down. 

From the below snapshot,  you can see the HPA thought the actual rps is 250m already, and set the pod number to 2, but then with low rps,  it tries to adjust the pod number to 4, 8, 11 again, and eventually to 2 again. 
<img width="1147" alt="image" src="https://user-images.githubusercontent.com/17263036/145600743-b0c42682-c004-457e-8567-0211cee70a67.png">

I want to understand why this behavior happens. 
Not sure whether I need to ask the question to HPA.  Please advise.

**Comment by zroubalik** [comment]
I recommend you to check HPA algorithm: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details

And if you would like to tweek the scaling behaviour, you can use this (you can find it under `ScaledObject.spec.advanced`):
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#configurable-scaling-behavior
